runtime/src/bank/tests.rs

```diff
@@ -12376,6 +12372,17 @@ where
     let bank = new_from_parent_next_epoch(bank, &bank_forks, 2);
     bank.freeze(); // trigger rent collection
 
+    // advance to the next slot to create a new bank, which is cleanable. The
+    // previous bank is at epoch boundary. An epoch boundary bank is not
+    // cleanable because epoch rewards sysvar is written in this slot. Before
+    // #3398, it was cleanable because rent collection will rewrite the epoch
+    // rewards sysvar due to rent_epoch not setting to u64::MAX. With #3398,
+    // rent_epoch is set to u64::MAX at creating time, so epoch rewards sysvar
+    // is not rewritten. Therefore, the slot at epoch boundary is now
+    // uncleanable. This is why we need to advance to the next slot.
```

Comment:
Without the PR,  epoch reward sysvar is created with rent_epoch = 0 at the epoch boundary slot X. And in a late slot Y, when rent collection runs, it finds that the rent_epoch is not MAX, so it write epoch reward sysvar at Y. This will make epoch reward sysvar dead in slot X. So clean can remove slot X. 

However, with this PR, epoch reward sysvar is created with rent_epoch = MAX at the epoch boundary slot X. In the late slot Y, when rent collection runs, it find that rent_epoch is already at MAX, so *no need* to write epoch rewards sysvar at Y. Therefore, the sysvar is still alive at slot X. Clean can't remove slot X. 

However, in the test we rely on clean to completely remove the slot if we overwrites the user account in a later slot. 
Because of the sysvar was implicitly created at the first slot of the epoch, which the test doesn't have control over, we should move our test of clean one slot later.

Reply:
I think it might be better not to refer to PR numbers in comments.

Reply:
yes. updated the comments.